compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -171,46 +173,83 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
                     | Res::Local(_) => true,
                     _ => false,
                 },
+                None,
             )
         } else {
             let item_span = path.last().unwrap().ident.span;
-            let (mod_prefix, mod_str) = if path.len() == 1 {
-                (String::new(), "this scope".to_string())
+            let (mod_prefix, mod_str, suggestion) = if path.len() == 1 {
+                debug!(
+                    "self.diagnostic_metadata.current_impl_items={:?}",
+                    self.diagnostic_metadata.current_impl_items
+                );
+                debug!(
+                    "self.diagnostic_metadata.current_function={:?}",
+                    self.diagnostic_metadata.current_function
+                );
+                let suggestion = if let Some(items) = self.diagnostic_metadata.current_impl_items
+                    && let Some((fn_kind, _)) = self.diagnostic_metadata.current_function
+                    && self.current_trait_ref.is_none()
+                    && let Some(function_ctxt) = fn_kind.ctxt()
+                    && let FnCtxt::Assoc(_) = function_ctxt
+                    && let Some(item) = items.iter().find(|i| {
+                        if let AssocItemKind::Fn(fn_) = &i.kind
+                            && fn_.sig.decl.inputs.is_empty()
```

Comment:
why do you only suggest that if there are no inputs?

I am not too familiar with this function

Reply:
I fixed it to suggest if inputs don't include `self`, `&self`, or the `&mut self`.